include/swift/AST/DiagnosticsSema.def

```diff
@@ -3570,6 +3570,9 @@ ERROR(self_assignment_prop,none,
 ERROR(property_use_in_closure_without_explicit_self,none,
       "reference to property %0 in closure requires explicit use of 'self' to"
       " make capture semantics explicit", (Identifier))
+WARNING(member_assign_to_closure_without_explicit_self,none,
+      "assignment of member %0 to closure requires explicit use of 'self' to"
```

Comment:
This warning is somewhat contradictory. We cannot require `self` but also warn about it. This should either be an error or there must be a way to silence this warning with syntax, which is... not the most desirable thing.

Reply:
If you wish to make this an error, it will require approval from swift evolution since this will break source.

Reply:
I feel that this is too small of a feature to break source-compatibility. Using closures this way could surely cause memory leaks; however, specifying self explicitly doesn't seem to scream that the self is being captured.

I am open to whatever seems best from your experience. In my mind, wrapping the function assignment with a closure seems like a good fixit, but we lose the conciseness.

```swift
sr8536_closure = baz
```

to

```swift
sr8536_closure = { [weak self] in
self?.baz()
}
```